test_workspace/Queue.bosatsu

```diff
@@ -78,6 +83,32 @@ eq_li = eq_List(eq_Int)
 
 q12 = empty.push(1).push(2)
 
+def samp(r)(fn): flat_map_Rand(r, fn)
```

Comment:
this isn't great. Either we should be really good at optimizing this to remove the uncurrying or I think bias to the uncurried desugaring in the first place.